src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(getClass())) {
+            Rectangle rectangle = (Rectangle) object;
+            return width.equals(rectangle.width)
+                    && length.equals(rectangle.length)
+                    && color.equals(rectangle.color);
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int generatedHashCode = 9;
+        generatedHashCode = generatedHashCode * 19 + width;
+        generatedHashCode = generatedHashCode * 19 + length;
+        if (color == null) {
+            generatedHashCode = generatedHashCode * 19;
+        }
+        generatedHashCode = generatedHashCode * 19 + color.hashCode();
```

Comment:
did you test your current solution with color = null?

Reply:
Oh, missed that, have already fixed!